components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/MeApiServiceImpl.java

```diff
@@ -9,31 +13,86 @@
+import org.wso2.charon3.core.protocol.endpoints.MeResourceManager;
 import org.wso2.msf4j.formparam.FormDataParam;
 import org.wso2.msf4j.formparam.FileInfo;
 import org.wso2.msf4j.Request;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.SecurityContext;
 
 public class MeApiServiceImpl extends MeApiService {
+    private static final Log LOG = LogFactory.getLog(GroupsApiServiceImpl.class);
+
     @Override
     public Response meDelete( Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.delete(userUniqueId, userManager);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
     @Override
     public Response meGet( Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.get(userUniqueId, userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response mePost( Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
-    }
+    public Response mePost(UserDTO body, Request request) throws NotFoundException {
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.create(body.toString(), userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);